hibernate-core/src/main/java/org/hibernate/query/sqm/internal/TypecheckUtil.java

```diff
@@ -140,6 +141,17 @@ private static boolean areTypesComparable(
 			return areEntityTypesComparable( (EntityType<?>) lhsType, (EntityType<?>) rhsType, factory );
 		}
 
+		if ( lhsType instanceof EntityType && rhsType instanceof PluralPersistentAttribute<?,?,?> ) {
+			if ( ((PluralPersistentAttribute<?,?,?>) rhsType).getElementType() instanceof EntityType ) {
+				return areEntityTypesComparable( (EntityType<?>) lhsType, (EntityType<?>) ((PluralPersistentAttribute<?,?,?>) rhsType).getElementType(), factory );
+			}
+		}
+		else if (rhsType instanceof EntityType && lhsType instanceof PluralPersistentAttribute<?,?,?>) {
+			if ( ((PluralPersistentAttribute<?,?,?>) lhsType).getElementType() instanceof EntityType ) {
+				return areEntityTypesComparable( (EntityType<?>) rhsType, (EntityType<?>) ((PluralPersistentAttribute<?,?,?>) lhsType).getElementType(), factory );
+			}
+		}
+
```

Comment:
Instead of this change, IMO you should rather use `pluralPath.getReferencedPathSource().getElementType()` when calling `assertComparable` in `SqmMemberOfPredicate`.

Reply:
`org.hibernate.query.sqm.tree.domain.SqmPluralValuedSimplePath#getReferencedPathSource` returns a `PluralPersistentAttribute` which has a `getElementType()` method.